api/src/test/java/io/kafbat/ui/service/rbac/RbacConnectorPermissionsTest.java

```diff
@@ -0,0 +1,258 @@
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.ReactiveSecurityContextHolder;
+import org.springframework.security.core.context.SecurityContext;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+/**
+ * Test cases for connector-level permissions in Kafka Connect.
+ * Tests the hierarchical permission system where connector-level permissions
+ * take precedence over connect-level permissions.
+ */
+@ExtendWith(MockitoExtension.class)
+class RbacConnectorPermissionsTest {
+
+  public static final String DEV_ROLE_NAME = "dev_role";
+  public static final String ADMIN_ROLE_NAME = "admin_role";
+  public static final String CLUSTER_NAME = "local";
+  public static final String CONNECT_NAME = "kafka-connect";
+  public static final String CONNECTOR_NAME = "my-connector";
+  public static final String ANOTHER_CONNECTOR_NAME = "another-connector";
+
+  private AccessControlService accessControlService;
+
+  @Mock
+  SecurityContext securityContext;
+
+  @Mock
+  Authentication authentication;
+
+  @Mock
+  RbacUser user;
+
+  @Mock
+  org.springframework.core.env.Environment environment;
+
+  @BeforeEach
+  void setUp() {
+    List<Role> roles = List.of(
+        getDevRole(),
+        getAdminRole()
+    );
+    RoleBasedAccessControlProperties properties = mock();
+    when(properties.getRoles()).thenReturn(roles);
+
+    accessControlService = new AccessControlService(null, properties, environment);
+    accessControlService.init();
+
+    when(securityContext.getAuthentication()).thenReturn(authentication);
+    when(authentication.getPrincipal()).thenReturn(user);
+  }
+
+  public void withSecurityContext(Runnable runnable) {
+    try (MockedStatic<ReactiveSecurityContextHolder> ctxHolder = Mockito.mockStatic(
+        ReactiveSecurityContextHolder.class)) {
+      // Mock static method to get security context
+      ctxHolder.when(ReactiveSecurityContextHolder::getContext)
+          .thenReturn(Mono.just(securityContext));
+      runnable.run();
+    }
+  }
+
+  /**
+   * Test that a user with specific connector-level permission can view the connector.
+   */
+  @Test
+  void validateAccess_withConnectorLevelPermission_allowed() {
+    withSecurityContext(() -> {
+      when(user.groups()).thenReturn(List.of(DEV_ROLE_NAME));
+      AccessContext context = AccessContext.builder()
+          .cluster(CLUSTER_NAME)
+          .connectActions(CONNECT_NAME, ConnectAction.VIEW)
+          .operationParams(Map.of("connectorName", CONNECTOR_NAME))
```

Comment:
But in current implementation connectorName would be missed in the controller validation process